lib/sla.js

```diff
@@ -45,80 +49,113 @@ function designateMixingBridge(client, bridgeName) {
+        .deviceStates));
+  return getDeviceState({deviceName: 'Stasis:' + name}).then(function (ds) {
+    return ds.state;
+  });
+}
+
+/**
+ * Adds the caller channel and originated channel (if applicable) to the bridge.
+ * @param {Object} client - the client (used for hanging up at the end).
+ * @param {Object} channel - Incoming channel to be added to the bridge.
  * @param {Object} dialed - Created/dialed channel to be added as well
  * @param {Object} bridge - Bridge that channels are to be added to.
+ * @param {Object} extension - the object data related to this sharedExtension
  * @return {Q} Q promise object.
  */
-function addChannelsToBridge(client, inbound, dialed, bridge) {
+function addChannelsToBridge(client, channel, dialed, bridge, extension) {
+  // Resets listeners, so that redundant events aren't accepted.
+  bridge.removeAllListeners('ChannelEnteredBridge');
+  bridge.removeAllListeners('ChannelLeftBridge');
   var defer = Q.defer();
-  var numInBridge = 0;
 
   console.log('Adding channels to bridge %s', bridge.id);
 
   var addChannel = Q.denodeify(bridge.addChannel.bind(bridge));
-  addChannel({channel: [inbound.id, dialed.id]})
+  addChannel({channel: channel.id})
     .catch(function (err) {
       defer.reject(err);
     });
 
+  if (dialed) {
+    addChannel({channel: dialed.id})
+      .catch(function (err) {
+        defer.reject(err);
+      });
+  }
+  bridge.on('ChannelEnteredBridge', channelEntered);
+  bridge.on('ChannelLeftBridge', channelLeft);
+  return defer.promise;
+
+  /**
+   * Function that displays a helpful message when a channel enters a bridge
+   * @param {Object} event - the event related to this bridge entering
+   * @param {Object} object - contains the bridge and channel in question
+   */
   function channelEntered(event, object) {
     console.log('Channel %s has entered the bridge', object.channel.id);
-    numInBridge += 1;
   }
 
-  bridge.on('ChannelEnteredBridge', channelEntered);
-
+  /**
+   * Function that is utilized when a channel leaves the bridge.
+   *   If the channel is a non-station channel, then hangs up remaining
+   *   channels. If the channel is a station, checks if the only non-station
+   *   channel is the only left, and then hangs up that non-station channel.
+   * @param {Object} event - the event related to the bridge leaving
+   * @param {Object} object - contains the bridge and channel in question
+   */
   function channelLeft(event, object) {
-    numInBridge -= 1;
-    console.log('Channel %s has left the bridge', object.channel.id);
-
-    if (numInBridge === 1) {
-      if (object.channel.id === inbound.id) {
-        console.log('Hanging up dialed channel %s', dialed.id);
-        var hangupDialed = Q.denodeify(dialed.hangup.bind(dialed));
-        hangupDialed()
+    if (isStation(extension, object.channel)) {
+      if (object.bridge.channels.length === 1) {
+        var hangupOutsider = Q.denodeify(client.channels.hangup.bind
```

Comment:
This was not addressed. As Ashley pointed out, this variable name is confusing and does not really indicate what this is hanging up by the addition of the name 'Outsider'. I would recommend calling it hangup to match the name of the function being denodeified.

Also, since you are modifying this, binding to client.channels will not work. You should bind to client. Since you have a channel instance however (object.channel), you can just bind to that and then you won't have to pass a channelId parameter to the call.

My guess is that this did not work so you added a channelId parameter to get around the issue.